### PR TITLE
ARROW-5054: [Release][Flight] Test Flight in Linux/macOS release verification scripts

### PR DESCRIPTION
**Pull request by wesm**

### CODE REVIEW

**Comment by kou** [comment]
I don't want to split the verification script.
It may cause "`test-release-verification.sh" works but `verify-release-candidate.sh` doesn't work" because of poor modularity feature of shell such as no namespace and easy to use global variables.
I like `TARBALL=/path/to/apache-arrow.tar.gz dev/release/verify-release-candidate.sh source ...` with:

```shell
if [ -z "${TARBALL:-}" ]; then
  fetch_archive ...
else
  tar xf ${TARBALL}
fi
```

And we need some work for GLib. We can't use `configure` with archive created by `git archive`.

Can I take over this?

**Comment by wesm** [comment]
@kou yes, you can take this over

**Comment by kou** [comment]
Thanks. I'll work on this.

File: dev/release/verify-release-candidate.sh [hunk 1]
```diff
@@ -482,13 +458,12 @@ if [ "$ARTIFACT" == "source" ]; then
   if [ ${TEST_INTEGRATION} -gt 0 ]; then
     test_integration
   fi
-  if [ ${TEST_GO} -gt 0 ]; then
-    test_go
-  fi
```
Comment (wesm) [inline dev/release/verify-release-candidate.sh]: This was removed by accident

File: dev/release/verify-release-candidate.sh [hunk 2]
```diff
@@ -370,15 +386,6 @@ test_rust() {
...
 
-test_go() {
-  pushd go
-
-  go get -v ./...
-  go test ./...
-
-  popd
-}
-
```
Comment (wesm) [inline dev/release/verify-release-candidate.sh]: Needs to be restored

**Review by wesm — COMMENTED** [review]
Minor issue with the Go tests being reverted from the script. Will restore and merge

**Comment by kou** [comment]
Oh, sorry. I'll restore them.

This is not completed yet. I'm still testing this.

**Comment by kou** [comment]
I've tested the followings:

  * `TEST_DEFAULT=0 TEST_INTEGRATION=1`
  * `TEST_DEFAULT=0 TEST_RUST=1`
  * `TEST_DEFAULT=0 TEST_GO=1`
  * `TEST_DEFAULT=0 TEST_RUBY=1`

They cover all source verification.

**Comment by wesm** [comment]
OK, great, thanks @kou!

**Comment by wesm** [comment]
+1

**Pull request closed without merging**